rasa/nlu/training_data/training_data.py

```diff
@@ -125,22 +182,78 @@ def sort_regex_features(self) -> None:
             self.regex_features, key=lambda e: "{}+{}".format(e["name"], e["pattern"])
         )
 
-    def as_json(self, **kwargs: Any) -> Text:
+    def fill_response_phrases(self):
+        """Set response phrase for all examples by looking up NLG stories"""
+        for example in self.training_examples:
+            response_key = example.get(MESSAGE_RESPONSE_KEY_ATTRIBUTE)
+            if response_key:
```

Comment:
should we do / log something if this key is not found?

Reply:
No, if `response_key` is None, that means the corresponding intent is not a retrieval intent and hence no response text needs to be fetched.
Adding this as a comment in the code.